src/main/java/com/boot/cart/service/CartService.java

```diff
@@ -1,255 +1,225 @@
+            } else {
+                CartEntry newCartEntry = new CartEntry();
+                newCartEntry.setProductName(productName);
+                newCartEntry.setQuantity(quantity);
+                newCartEntry.setCart(cart);
+                cartEntries.add(newCartEntry);
+            }
         }
 
-        cart.setUser(user);
-
         for (int i = 0; i < quantity; i++) {
-            productList.add(product);
-
-            productTotal += product.getProductPrice();
+            productTotal += productDTO.getPrice();
         }
 
-        cart.setProductList(productList);
+        cart.setEntries(cartEntries);
         cart.setTotal(cart.getTotal() + productTotal);
+        cart.setUserId(user.getId());
 
-        cartRepository.save(cart);
-
-        return CartMapper.cartEntityToDto(cart);
+        ;
 
+        return cartEntityToDto(cartRepository.save(cart));
     }
 
     public CartDTO updateProductFromCart(String email, String productName, int quantity)
             throws InvalidInputDataException, EntityNotFoundException {
+        log.info("removeProductFromCart - process started");
 
-        Product product;
+        ProductDTO productDTO;
         try {
-            product = ProductMapper.DtoToProductEntity(productServiceClient.callGetProductByProductName(productName));
+            productDTO = productServiceClient.callGetProductByProductName(productName);
         } catch (HttpClientErrorException.NotFound e) {
             throw new EntityNotFoundException("Product: " + productName + " not found in the Database!");
         }
 
-        User user;
+        if (productDTO.getStock() == 0) {
+            throw new InvalidInputDataException("We are sorry, but currently: " + productName + " is out of order!");
+        }
+
+        if (productDTO.getStock() < quantity) {
+            throw new InvalidInputDataException("You can not remove more than: " + productDTO.getStock() + " "
+                    + productName + " Products from your shopping cart!");
+        }
+        UserDTO user;
         try {
-            user = UserMapper.DtoToUserEntity(userServiceClient.callGetUserByEmail(email));
+            user = userServiceClient.callGetUserByEmail(email);
         } catch (HttpClientErrorException.NotFound e) {
-            throw new EntityNotFoundException("UserName: " + email + " not found in the Database!");
+            throw new EntityNotFoundException("Email: " + email + " not found in the Database!");
         }
-        Cart cart = cartRepository.findByUser(user);
-        List<Product> productList = cart.getProductList();
-
-        Long productsInCart = productList.stream().filter(p -> p.getProductName().equals(productName)).count();
 
-
-
-        if (productsInCart == 0) {
-            throw new InvalidInputDataException(
-                    "You currently have " + productsInCart + " Products: " + productName + " in cart!");
+        Cart cart = cartRepository.findByUserId(user.getId());
```

Comment:
a small code improvement to this would be to do something like:
Cart cart = cartRepository.findByUserId(user.getId).orElseThrow(() -> throw new EntityNotFoundException("Cart not found in the Database!"));

Reply:
Done! looks waaay more cleaner this way!
I have to implement this change for all Repositoty methods used in all projects.. don't I ? 🤣 🤣 🤣

Reply:
of course 😄 you elevate your coding standard 😉